core/blockchain.go

```diff
@@ -1369,6 +1376,15 @@ func (self *BlockChain) reorg(oldBlock, newBlock *types.Block) error {
 		go self.eventMux.Post(RemovedLogsEvent{deletedLogs})
 	}
 
+	if len(oldChain) > 0 {
+		fmt.Println("side chain length =", len(oldChain))
+		go func() {
+			for _, block := range oldChain {
+				self.eventMux.Post(ChainSideEvent{Block: block, Logs: deletedLogsByHash[block.Hash()]})
+			}
+		}()
```

Comment:
Grouping of the goroutines and possibly the `len` check is done in another PR. Want to keep the changes to a minimum.